mypyc/genops.py

```diff
@@ -435,11 +437,32 @@ def prepare_method_def(ir: ClassIR, module_name: str, cdef: ClassDef, mapper: Ma
         decl = prepare_func_def(module_name, cdef.name, node.func, mapper)
         if not node.decorators:
             ir.method_decls[node.name()] = decl
+        elif isinstance(node.decorators[0], MemberExpr) and node.decorators[0].name == 'setter':
+            # Make property setter name different than getter name so there are no
+            # name clashes when generating C code, and property lookup at the IR level
+            # works correctly.
+            decl.name = PROPSET_PREFIX + decl.name
+            ir.method_decls[PROPSET_PREFIX + node.name()] = decl
+
         if node.func.is_property:
             assert node.func.type
             ir.property_types[node.name()] = decl.sig.ret_type
 
 
+def check_multipart_property_def(prop: OverloadedFuncDef) -> None:
```

Comment:
While we are not perfect about this yet, we are striving to produce error messages and continue compilation (so that we can produce a full set of error messages, instead of stopping early) instead of failing assertions when we hit a case we can't compile.

So this should become a function that returns whether it is a valid property decorator and we should check it and emit an error message if it isn't and not crash. Then add a test for that to `testErrorOutput` in `commandline.test`.